src/plugin/context/mod.rs

```diff
@@ -321,14 +354,40 @@ impl RapierContext {
                         &mut self.ccd_solver,
                         None,
                         hooks,
-                        events,
+                        event_handler,
                     );
                 }
             }
         }
+        if let Some(mut event_queue) = event_queue {
+            // SAFETY: event_queue and its inner locksare only accessed from
+            // within `self.pipeline.step` called above, so we can unwrap here safely.
+            self.collision_events_to_send =
+                std::mem::take(event_queue.collision_events.get_mut().unwrap());
+            self.contact_force_events_to_send =
+                std::mem::take(event_queue.contact_force_events.get_mut().unwrap());
+        }
+    }
+    /// Generates bevy events for any physics interactions that have happened
+    /// that are stored in the events list
+    pub fn send_bevy_events(
+        &mut self,
+        collision_event_writer: &mut EventWriter<CollisionEvent>,
+        contact_force_event_writer: &mut EventWriter<ContactForceEvent>,
+    ) {
+        for collision_event in self.collision_events_to_send.iter() {
+            collision_event_writer.send(*collision_event);
+        }
+        self.collision_events_to_send.clear();
+
+        for contact_force_event in self.contact_force_events_to_send.iter() {
```

Comment:
self.contact_force_events_to_send.drain(..) could be used to make usage more clear here, and same with the above for loop.